homeassistant/components/integration/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for Integration - Riemann sum integral integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_METHOD,
+    CONF_NAME,
+    TIME_DAYS,
+    TIME_HOURS,
+    TIME_MINUTES,
+    TIME_SECONDS,
+)
+from homeassistant.helpers import selector
+from homeassistant.helpers.helper_config_entry_flow import (
+    HelperConfigFlowHandler,
+    HelperFlowStep,
+)
+
+from .const import (
+    CONF_ROUND_DIGITS,
+    CONF_SOURCE_SENSOR,
+    CONF_UNIT_PREFIX,
+    CONF_UNIT_TIME,
+    DOMAIN,
+    INTEGRATION_METHODS,
+    METHOD_TRAPEZOIDAL,
+)
+
+UNIT_PREFIXES = ["none", "k", "M", "G", "T"]
+TIME_UNITS = [TIME_SECONDS, TIME_MINUTES, TIME_HOURS, TIME_DAYS]
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_METHOD, default=METHOD_TRAPEZOIDAL): selector.selector(
+            {"select": {"options": INTEGRATION_METHODS}}
+        ),
+        vol.Required(CONF_ROUND_DIGITS, default=2): selector.selector(
+            {"number": {"min": 0, "max": 6, "mode": "box"}}
+        ),
+        vol.Required(CONF_UNIT_PREFIX, default=TIME_HOURS): selector.selector(
+            {"select": {"options": UNIT_PREFIXES}}
+        ),
```

Comment:
<img width="393" alt="image" src="https://user-images.githubusercontent.com/195327/158818938-e8e9cf4c-54ce-494a-98bb-1124994aafeb.png">

Lets use a key/value pairs in this selector, to make the input Friendlier